Source/itemdat.h

```diff
@@ -13,7 +13,7 @@
 namespace devilution {
 
 /** @todo add missing values and apply */
-enum _item_indexes : int16_t {
+enum _item_indexes : int16_t { // TODO defines all indexes in AllItemsList

```

Comment:
Are negative values expected to correspond to valid Item IDs? If not using an unsigned type for the backing type would save having to do `<0` checks, letting that be handled by `>= LAST`

Reply:
-1 == no item

Reply:
-1 isn't considered a valid item though, `AllItemsList[-1]` is an out of bounds access. Looking at the way `Item.IDidx` is used it seems like `Item._itype == None` is used as the guard, then `IDidx` is used as a straight index into that array.